src/engine/strat_engine/engine.rs

```diff
@@ -124,6 +140,51 @@ impl Engine for StratEngine {
         Ok(uuid)
     }
 
+    /// Evaluate a device node & devicemapper::Device to see if it's a valid
+    /// stratis device.  If all the devices are present in the pool and the pool isn't already
+    /// up and running, it will get setup and the pool uuid will be returned.
+    fn block_evaluate(&mut self,
+                      dev_node: PathBuf,
+                      device: Device)
+                      -> EngineResult<Option<PoolUuid>> {
+        let mut rc = None;
+        let mut duplicate = false;
+
+        if let Some(pool_uuid) = is_stratis_device(&dev_node)? {
+
+            self.incomplete_pools
+                .entry(pool_uuid)
+                .or_insert_with(HashMap::new)
+                .insert(device, dev_node);
+
+            // Check to see if we already have a pool up and running with this uuid,
+            // if we do we are going to ignore this setup.  We can re-create this
+            // case by creating synthetic "add" events for existing stratis devices.
+            if !self.pools.contains_uuid(pool_uuid) {
+
+                let devices = self.incomplete_pools
+                    .get(&pool_uuid)
+                    .expect("We previously inserted pool uuid!");
+
+                if let Ok(pool_setup) = StratPool::setup(pool_uuid, devices) {
+                    self.pools.insert(pool_setup);
+                    rc = Some(pool_uuid);
+                }
+            } else {
+                // We have this pool already so we shall ignore.
+                warn!("Pool {} is already known, ignoring!", pool_uuid);
```

Comment:
My new official policy is that I will not object to debug output of this specific sort, i.e., a notification of a problem that cannot be better encapsulated in the return value, as we would want to log this stuff in some way. The log message serves as a marker of our future, better, intentions. You get to be the first beneficiary of this policy :-).